source/OptimaJet.DWKit.StarterApplication/Forms/Products/BaseProductForm.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Linq;
+using Optimajet.DWKit.StarterApplication.Models;
+using OptimaJet.DWKit.StarterApplication.Repositories;
+using OptimaJet.DWKit.StarterApplication.Services;
+
+namespace OptimaJet.DWKit.StarterApplication.Forms.Products
+{
+    public class BaseProductForm : BaseForm
+    {
+        public BaseProductForm(
+            UserRepository userRepository,
+            ICurrentUserContext currentUserContext) : base(userRepository, currentUserContext)
+        {
+        }
+        protected void ValidateProduct(Project project, ProductDefinition productDefinition)
+        {
+            if ((project == null) || (productDefinition == null))
+            {
+                // Allowing it to return in these cases should cause the base to hit normal
+                // foreign key failures
+                return;
+            }
+            if ((project.Organization == null))
```

Comment:
since this is a product form, should we assume that a project has already been validated / has already been persisted to the database? if so, we may be able to skip this check.

Reply:
Removed the check